zipline/utils/sentinel.py

```diff
@@ -27,14 +41,14 @@ def __deepcopy__(self, _memo):
         def __copy__(self):
             return self
 
-    cls = type(result)
-    cls.__name__ = name
+    cls = type(Sentinel)
     try:
         # traverse up one frame to find the module where this is defined
-        cls.__module__ = sys._getframe(1).f_globals.get(
-            '__name__',
-            '__main__',
-        )
-    except (AttributeError, ValueError):
-        pass
-    return result
+        cls.__module__ = sys._getframe(1).f_globals['__name__']
+    except (AttributeError, ValueError, KeyError):
```

Comment:
Yeah, if someone called this function from a c library that was using libpython but didn't already have a python stack setup.